### ISSUE TITLE
Type definitions and intellisense

### ISSUE THREAD

**Issue by neefrehman** [mattdesl/canvas-sketch-util#13]
Hi @mattdesl, I've just been watching your Introduction to creative coding class on Frontend Masters, and noticed that when you're using this library your editor is giving you autocomplete and type annotations (pictured below). I've been looking around and can't see any type definitions in the library, is there a different way that you're getting these?

![image](https://user-images.githubusercontent.com/29544565/85930432-49ccbd00-b8b4-11ea-9794-775dcd88e4b2.png)

**Comment by awaer** [comment]
I believe these are generated by Visual Studio Code.

https://code.visualstudio.com/docs/editor/intellisense

**Comment by neefrehman** [comment]
@awaer yes they should be generated by the editor, but usually this is only possible if the library emits a `.d.ts` file, which I don't think canvas-sketch-util does. I'm using VSCode but can't see any of the definitions. Have you had luck getting them to show?

**Comment by mattdesl** [comment]
I'm definitely :+1: on Typescript definitions. It's not really an area I'm familiar with, but open to PRs and more discussion.

**Comment by smhutch** [comment]
@mattdesl I have been using these utils quite extensively in my own projects, and have made a start on adding types to my [personal site](https://github.com/smhutch/smhutch/tree/main/typings/canvas-sketch-util). I would be happy to help out with adding types to this repo.

**Comment by elesq** [comment]
was this ever resolved ?

**Comment by bulkan** [comment]
@smhutch can you provide a PR with your typedefs?

**Comment by elesq** [comment]
Did this ever get followed up on?

**Comment by babausias** [comment]
I found this package which has enabled intellisense in the IDE for me:
https://www.npmjs.com/package/canvas-sketch-util-types

I added it manually to the dev dependencies to the package.json:
```json
"devDependencies": {
    "@types/canvas-sketch-util": "npm:canvas-sketch-util-types"
  }
```
